### PR TITLE
Move scripts out of lib/tool_shed.

### PR DESCRIPTION
**Pull request by davebx**

### CODE REVIEW

**Comment by jmchilton** [comment]
The minor tag means it changes made by devteam members to features released this cycle - it doesn't apply here.

File: scripts/tool_shed/api/common.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import urllib
 import urllib2
 
-sys.path.insert(1, os.path.join( os.path.dirname( __file__ ), os.pardir, os.pardir, os.pardir ) )
+sys.path.insert( 1, 'lib' )
```
Comment (nsoranzo) [inline scripts/tool_shed/api/common.py]: This should be

```
sys.path.insert(1, os.path.join(os.path.dirname( __file__ ), os.pardir, os.pardir, os.pardir, 'lib'))
```

File: scripts/tool_shed/api/export.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 import tempfile
 import urllib2
 
-sys.path.insert(1, os.path.join( os.path.dirname( __file__ ), os.pardir, os.pardir, os.pardir ) )
+sys.path.insert( 1, 'lib' )
```
Comment (nsoranzo) [inline scripts/tool_shed/api/export.py]: Same as above.

File: scripts/tool_shed/api/get_filtered_repository_revisions.py [hunk 3]
```diff
@@ -25,7 +25,7 @@
 import sys
 import urllib
 
-sys.path.insert(1, os.path.join( os.path.dirname( __file__ ), os.pardir, os.pardir, os.pardir ) )
+sys.path.insert( 1, 'lib' )
```
Comment (nsoranzo) [inline scripts/tool_shed/api/get_filtered_repository_revisions.py]: Same as above.

**Comment by nsoranzo** [comment]
`sys.path.insert()` should be updated also in `scripts/tool_shed/bootstrap_tool_shed/create_user_with_api_key.py` and the other Python files following it in the list of moved files.

**Comment by davebx** [comment]
@jmchilton Where is this documented? I should probably bookmark that page.

**Comment by martenson** [comment]
@davebx https://github.com/galaxyproject/galaxy/blob/dev/doc/source/project/issues.rst or https://docs.galaxyproject.org/en/master/project/issues.html

**Comment by jmchilton** [comment]
:+1: It wasn't really obvious to me at first that scripts belong in `scripts/` and not `lib/` but I think these do since they mess with sys.path. I think as a design goal - nothing in `lib/` should be messing with `sys.path` - all modules in `lib` should be importable, side-effect free, without errors if non-optional dependencies are available.

**Comment by davebx** [comment]
@jmchilton In addition to the above, with which I agree, I don't see `lib/` being appropriate for scripts run from the command line, nor `scripts/` being appropriate for modules loaded into galaxy, the toolshed, or whichever app one is dealing with.

**Comment by nsoranzo** [comment]
@davebx sys.path.insert() still need to be updated in many Python files moved in `scripts/tool_shed/`:
scripts/tool_shed/check_filesystem_for_empty_tool_dependency_installation_paths.py
scripts/tool_shed/check_repositories_for_functional_tests.py
scripts/tool_shed/check_s3_for_empty_tool_dependency_installation_paths.py
scripts/tool_shed/check_tool_dependency_definition_repositories.py
scripts/tool_shed/deprecate_repositories_without_metadata.py
scripts/tool_shed/show_tool_dependency_installation_dir_contents.py

**Comment by martenson** [comment]
Is this consistent now? @nsoranzo @davebx 

this seems to have caused https://github.com/galaxyproject/galaxy/pull/2121

**Pull request merged**